### ISSUE TITLE
Add support for additional (and custom) services

### ISSUE THREAD

**Issue by mreithub** [ondevice/ondevice#7]
Right now the only officially supported protocol is SSH (on the client - the cloud service is protocol agnostic, so in principle you can already tunnel anything you want).

It would be nice to have ways to tunnel things like HTTP/HTTPS directly (e.g. to give people access to your development web server or to access maintenance apps/web interfaces) without having to tunnel them through SSH.

Ideally, people should be able to expose arbitrary services (i.e. any TCP ip:port combination), but would have to explicitly configure the ip+port when setting up the service.

## Example syntax

The following commandline examples are just ideas on how certain functionality could be implemented.  
They might be entirely unrealistic or impractical (x11vnc for example would only work on linux based devices, it's not the best idea to tunnel UDP over TCP - to just name a few issues)

```bash
ondevice service add myRedis tcp://localhost:6379/
ondevice service add dns udp://8.8.8.8:53/
ondevice service add echo echo://
ondevice service add someApp http://localhost:8080/someApp/
ondevice service add vnc vnc:// # could start an x11vnc instance. To connect to existing VNC servers, 'tcp://...:590x' could be used instead

# list enabled services
ondevice service #or
ondevice service list

# remove a service
ondevice service rm $name
```

For security reasons, enabling services should only be possible directly on the device (i.e. from the command line, so either locally or via SSH).

Especially for HTTP it might be useful to add support for multichannel tunnels (since most browsers will maintain several connections to a server at once and each handshake is costy compared to an already-open connection).

On the client side, the respective commands for above services could be:

```bash
ondevice :tunnel $devId myRedis $localPort
ondevice :udp $devId dns 5353
echo hello | ondevice :pipe $devId echo
ondevice curl http://someApp.$devId/ # gotta think about that one - there might be a few caveats to overcome
ondevice vnc devId # would have to try to find a local VNC client
```

`:tunnel` and `:udp` would work similar to the already existing `ondevice :pipe` (which is still not considered API-stable and thus left undocumented for now).

Note: Depends on #1

**Comment by mreithub** [comment]
`ondevice :tunnel` could have something like a `--persistent` or `--reconnect` parameter to keep the listener open instead of exiting after the connection was closed (similar to what autossh does)
